src/Conf.ml

```diff
@@ -893,9 +897,21 @@ let kind_of fname =
   | ".mlt" -> `Use_file
   | _ -> !kind
 
+let update_using_xdg =
+  try
+    let filename =
+      Filename.concat
+        (Caml.Sys.getenv "XDG_CONFIG_HOME")
+        "ocamlformat/.ocamlformat"
+    in
+    read_config ~filename ~parents:false
+  with
+  | Caml.Not_found | Not_found_s _ -> Fn.id
```

Comment:
It looks like this code is intentionally reading the env only once at the beginning. Could you make this clearer using `Staged`.